k9mail/src/main/java/com/fsck/k9/Preferences.java

```diff
@@ -164,4 +164,17 @@ public void setDefaultAccount(Account account) {
     public SharedPreferences getPreferences() {
         return mStorage;
     }
+
+    public static <T extends Enum<T>> T getEnumStringPref(SharedPreferences prefs, String key, T defaultEnum) {
+        String stringPref = prefs.getString(key, defaultEnum.name());
+        try {
+            return Enum.valueOf(defaultEnum.getDeclaringClass(), stringPref);
+        } catch (Exception ex) {
```

Comment:
i know, but moving/touching code is always an opportunity to clean things up. the best solution would be to make the null check explicit:

```java
String stringPref = prefs.getString(key, null);
if (stringPref == null) {
    return defaultEnum;
} else {
    try {
        return Enum.valueOf(defaultEnum.getDeclaringClass(), stringPref);
    } catch (IllegalArgumentException ex) {
        Log.w(K9.LOG_TAG, "Unable to convert preference key [" + key +
                "] value [" + stringPref + "] to enum of type " + defaultEnum.getDeclaringClass(), ex);

        return defaultEnum;
    }
} 
```